### ISSUE TITLE
Date ~2 weeks ago incorrectly called `last week`

### ISSUE THREAD

**Issue by jaydenseric** [kapouer/duration-relativetimeformat#1]
Dates that are about two weeks apart incorrectly result in a `last week` duration:

```js
import createFormatDuration from 'duration-relativetimeformat';

const formatDuration = createFormatDuration('en-US');
const dateTo = new Date('2021-05-18T23:43:04.024Z');
const dateFrom = new Date('2021-06-01T02:09:26.441Z');

console.log(formatDuration(dateTo, dateFrom));
```

The words `last week` imply the date falls within the last 7 days.

**Comment by kapouer** [comment]
I updated the tests to make sure the change in behavior will go along a change of unit tests.

**Comment by kapouer** [comment]
Okay i suppose the transitions between "x units ago" and "x+1 units ago" has similar issues.
Maybe this would work better:
- change of units (second -> minute, minute -> hour, hour -> day, day -> week, week -> month, month -> year) should happen when duration is one unit before biggerUnit (>59s is like a minute, >59mn is like one hour, >6 days like a week, > 24 days like a month, > 11month like a year)
- change of amount within the same unit: should happen at half distance (1.45 unit is 1 unit, 1.55 unit is 2 units).

**Comment by jaydenseric** [comment]
It seems sketchy to say `1 week ago` if it was really 10 days ago. What if exactly 7 days = `1 week ago` and 12 days = `over 1 week ago`.

Thinking out loud about how a human would describe a relative time, what about a concept of max depth? So, with no max depth:

- `6 hours, 14 minutes, and 8 seconds ago`
- `1 week, 3 days, 2 hours, 3 minutes, and 34 seconds ago`

With max depth of 2:

- `6 hours and 14 minutes ago`
- `1 week and 3 days ago`

**Comment by kapouer** [comment]
The point is to let the intl be done entirely by Intl.RelativeTimeFormat, which AFAIK does not support "Over x units" or "Less than x units".

Adding a "maxDepth" option is doable, but is that really something useful ?
I don't see the point in showing "6 hours and 14 minutes ago" to the user.
I think a better approach would be this:

- 23 hours -> 23 hours
- 23h30 < x < 24h30 -> day
- 36 hours -> 36 hours
- 47h < x < 49h -> 2 days
- 6d < x < 8d -> a week
- 9d -> 9 days
- etc...

**Comment by jaydenseric** [comment]
I don't think only 6.5 days should be called `a week ago`, a week is at least 7 days.

**Comment by kapouer** [comment]
Indeed, so between 7 days and 8 days (excluded) is a week, and likewise for all other units.

**Comment by kapouer** [comment]
Also i recognize that "36 hours" or "10 days" are okay, but "80 hours" or "52 months" are not.
The more the duration, the bigger the rounding: "80 hours" should be "4 days" and "52 months" 4 years.

maxDepth, on second though, is not doable: Intl.RelativeTimeFormat does not format multiple units at once, so there is no way to properly get the right translation for "x units *and* y subunits".

I think the right approach is to let the user decide *when* the duration is written using more precise units or not, by giving a "precision" parameter in percents:
- 90 seconds might be 1mn or 90s
- 3610 seconds however will be too precise and fallback to 1 hour
- 23 hours will always be 23 hours
- 25 hours might be 1 day or 25 hours depending on precision
- 49 hours might be 2 days or stay 49 hours

now i need to figure how to implement this :)

**Comment by jaydenseric** [comment]
I used `duration-relativetimeformat` in [`audit-age`](https://github.com/jaydenseric/audit-age) [v1.0.0](https://github.com/jaydenseric/audit-age/releases/tag/v1.0.0), you can see the durations in the output if you run `npx audit-age` in a project:

<img width="549" alt="Screen Shot 2021-06-04 at 2 43 18 pm" src="https://user-images.githubusercontent.com/1754873/120747307-29859380-c544-11eb-91cf-bbc3d03c9540.png">

It would look really nice once this issue is patched; if the pink `last week` would instead say `last month` to match the pink `Month` age category.

**Comment by kapouer** [comment]
Yes, i also found out that relativetime format can output "this day", "this month", "this week"...
I'll wrap my head around it and fix it tomorrow.

**Comment by kapouer** [comment]
I think this is fixed in 2.0.1.
